src/runtime/cache.ts

```diff
@@ -78,11 +78,13 @@ export function defineCachedFunction<T = any>(
 
     const _resolve = async () => {
       if (!pending[key]) {
-        // Remove cached entry to prevent using expired cache on concurrent requests
-        entry.value = undefined;
-        entry.integrity = undefined;
-        entry.mtime = undefined;
-        entry.expires = undefined;
+        if (entry.value && (opts.staleMaxAge || 0) >= 0) {
```

Comment:
(falsy values such as `0`, `false` and `null` can also be a value for result)

```suggestion
        if (entry.value !== undefined && (opts.staleMaxAge || 0) >= 0) {
```